src/controller/python/chip/clusters/Command.py

```diff
@@ -201,6 +283,61 @@ def SendCommand(future: Future, eventLoop, responseType: Type, device, commandPa
+                      suppressResponse: Union[None, bool] = None) -> PyChipError:
+    ''' Send a cluster-object encapsulated command to a device and does the following:
+            - On receipt of a successful data response, returns the cluster-object equivalent through the provided future.
+            - None (on a successful response containing no data)
+            - Raises an exception if any errors are encountered.
+
+        If no response type is provided above, the type will be automatically deduced.
+
+        If a valid timedRequestTimeoutMs is provided, a timed interaction will be initiated instead.
+        If a valid interactionTimeoutMs is provided, the interaction will terminate with a CHIP_ERROR_TIMEOUT if a response
+        has not been received within that timeout. If it isn't provided, a sensible value will be automatically computed that
+        accounts for the underlying characteristics of both the transport and the responsiveness of the receiver.
+    '''
+    handle = chip.native.GetLibraryHandle()
+
+    responseTypes = []
+    commandargs = []
+    for command in commands:
+        endpoint = command[0]
+        clusterCommand = command[1]
+        responseType = command[2]
+        if (responseType is not None) and (not issubclass(responseType, ClusterCommand)):
+            raise ValueError("responseType must be a ClusterCommand or None")
+        if clusterCommand.must_use_timed_invoke and timedRequestTimeoutMs is None or timedRequestTimeoutMs == 0:
+            raise chip.interaction_model.InteractionModelError(chip.interaction_model.Status.NeedsTimedInteraction)
+        commandPath = chip.interaction_model.CommandPathIBStruct.parse(
+            b'\x00' * chip.interaction_model.CommandPathIBStruct.sizeof())
+        commandPath.EndpointId = endpoint
+        commandPath.ClusterId = clusterCommand.cluster_id
+        commandPath.CommandId = clusterCommand.command_id
+        commandPath = chip.interaction_model.CommandPathIBStruct.build(commandPath)
+        payloadTLV = clusterCommand.ToTLV()
+
+        commandargs.append(c_char_p(commandPath))
+        commandargs.append(c_char_p(bytes(payloadTLV)))
+        commandargs.append(c_size_t(len(payloadTLV)))
+
+        responseTypes.append(responseType)
+
+    transaction = AsyncBatchCommandsTransaction(future, eventLoop, responseTypes)
+    ctypes.pythonapi.Py_IncRef(ctypes.py_object(transaction))
+
+    return builtins.chipStack.Call(
+        lambda: handle.pychip_CommandSender_SendBatchCommands(
+            py_object(transaction), device,
+            c_uint16(0 if timedRequestTimeoutMs is None else timedRequestTimeoutMs),
+            c_uint16(0 if interactionTimeoutMs is None else interactionTimeoutMs),
+            c_uint16(0 if busyWaitMs is None else busyWaitMs),
+            c_bool(False if suppressResponse is None else suppressResponse),
+            c_size_t(len(commands)), *commandargs)
```

Comment:
With large number of command, this is a very large varargs call... Something more scalable may be to encode a linear array of structs <const char*, const uint8_t, size_t> and parse it on the other side.

Reply:
I would argue this is possibly worse for Write Attributes as you can span multiple messages.

Do you mind if this is something that is addressed at some other time. I don't want to commit to it as I don't know the scope of the work, but I am assuming that